charts/athens-proxy/values.yaml

```diff
@@ -2,7 +2,7 @@ replicaCount: 1
 image:
   registry: docker.io
   repository: gomods/athens
-  tag: v0.3.0
+  tag: "{{ .Chart.AppVersion }}"
```

Comment:
I think we should avoid using templates in values files, since it makes it harder for users of this chart. Keeping `image.tag` and `appVersion` in sync manually is better for readability.

Reply:
why you think so?
it makes only one place to change the tag. it is used widely in helm charts

Reply:
add a comment e.g. `# Note that by default we use appVersion to set image tag` above the tag would make things easier to understand.
even the current chart version is not in sync between `appVersion` and `tag`

Reply:
Yes, I understand why you want to reuse `appVersion`. I'm just pointing out that this change makes it harder to reuse the helm chart in an enterprise environment where you have to use your own images and where you usually override `image.registry`, `image.repository` and `image.tag`. :)

Reply:
right, I see your point :-), let's use a bit different solution then changing files:
values.yaml
```
...
image:
  registry: docker.io
  repository: gomods/athens
  # Note that by default we use appVersion to get image tag
  # But you can set the tag if there are needs to overwrite it
  # tag:
....
```

deployment.yaml
```
...
image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
...
```

But default tag will be taken from `appVersion` and can be overwritten in the `values.yaml` as before

Reply:
@minhdanh can you make those ^^ changes, please? thanks

Reply:
It's ok for me, but there's another thing: https://github.com/gomods/athens/blob/master/charts/athens-proxy/templates/_helpers.tpl#L6

The image tag is used to decide the endpoint of livenessProbes and readinessProbes.
`v0.3.0` is hardcoded. Looks like the endpoints won't be rendered with image tag/appVersion  `v0.3.1`.

Reply:
remove https://github.com/gomods/athens/blob/master/charts/athens-proxy/templates/_helpers.tpl#L5-L10` if it is supported in v0.3.0 and onwards?

or move that part to the `values.yaml` with instructions to setup for all versions?